### ISSUE TITLE
"Click for texts" not displaying on web front-end

### ISSUE THREAD

**Issue by apdame** [Bookworm-project/BookwormDB#128]
So, I've got a bookworm [up here](http://www.mikedame.com:7531/TUArch/#), but any attempt to use the "Click for texts function" gives this Console error:

Uncaught SyntaxError: Unexpected token G in JSON at position 0
    at JSON.parse (<anonymous>)
    at Object.success (bookworm.gui.js:882)
    at i (jquery.min.js:2)
    at Object.fireWith [as resolveWith] (jquery.min.js:2)
    at z (jquery.min.js:4)
    at XMLHttpRequest.<anonymous> (jquery.min.js:4)

Is this a server-side issue, or with my input/jsoncatalog files? The usual method of trolling StackExchange hasn't provided much clarity.

**Comment by apdame** [comment]
Possibly related, I've also consistently getting this error during builds (on a CentOS 7 server with the latest MariaDB installed):

20/2017 20:34:42:ERROR:Query failed:
CREATE TABLE IF NOT EXISTS NewsgroupsDisk (
        bookid MEDIUMINT UNSIGNED,
         Newsgroups VARCHAR(255), INDEX (Newsgroups), INDEX (bookid, Newsgroups )
        );

Traceback (most recent call last):
  File "/usr/bin/bookworm", line 9, in <module>
    load_entry_point('bookwormDB==0.4.0', 'console_scripts', 'bookworm')()
  File "/usr/lib/python2.7/site-packages/bookwormDB/manager.py", line 567, in run_arguments
    getattr(my_bookworm,args.action)(args)
  File "/usr/lib/python2.7/site-packages/bookwormDB/manager.py", line 238, in prep
    getattr(self,args.goal)()
  File "/usr/lib/python2.7/site-packages/bookwormDB/manager.py", line 336, in database_metadata
    Bookworm.load_book_list()
  File "/usr/lib/python2.7/site-packages/bookwormDB/CreateDatabase.py", line 248, in load_book_list
    self.variableSet.loadMetadata()
  File "/usr/lib/python2.7/site-packages/bookwormDB/variableSet.py", line 764, in loadMetadata
    variable.buildDiskTable()
  File "/usr/lib/python2.7/site-packages/bookwormDB/variableSet.py", line 238, in buildDiskTable
    );""")
  File "/usr/lib/python2.7/site-packages/bookwormDB/CreateDatabase.py", line 111, in query
    cursor.execute(sql)
  File "/usr/lib64/python2.7/site-packages/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "/usr/lib64/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
_mysql_exceptions.OperationalError: (1071, 'Specified key was too long; max key length is 1000 bytes')
make: *** [.bookworm/targets/database_metadata] Error 1

**Comment by bmschmidt** [comment]
1. First thought: do you have a line specifying the searchstring in your `field_descriptions.json` file? Would look like this.

```
    {"datatype": "searchstring", "field": "searchstring", "unique": true, "type": "text"},
```


2. I haven't seen this particular error before, so it could be related, I guess. The reason would probably be if you have a class label over 255 characters long. (Maybe CentOS is allocating 4 bytes for a unicode character in the index? Usually it's 3, which is why this is new to me.) As a rough solution, you might try clipping all the newsgroup names at 100 characters or so and see if that fixes anything; if it does, there might be something more we could do.

**Comment by apdame** [comment]
1. I did, but at some point I must have started using an incorrect/older field_descriptions.json file, since the "unique" flag was set to false and the "type" flag was wrong as well. So, using the right string fixed that issue.

2. Whatever it was, I think it's a server-side issue. I never could figure it out, so I just worked around it by doing the build on an Ubuntu VM and porting everything over to the web server. It's [up and working](http://www.averydame.org:7531/TransUsenetArchive/#), though, so that's all that matters on my end.

Since everything seems to be solved, I'm going to go ahead and close the issue.
